src/net/java/sip/communicator/impl/protocol/jabber/extensions/colibri/ColibriConferenceIQ.java

```diff
@@ -2271,13 +2361,63 @@ public void setDisplayName(String displayName)
         }
 
         /**
+         * Sets the transport element of this <tt>ChannelBundle</tt>.
+         * @param transport the transport to set.
+         */
+        public void setTransport(IceUdpTransportPacketExtension transport)
+        {
+            this.transport = transport;
```

Comment:
`addExtension(transport);` (+ null handling, calling twice, etc.)
Saving transport to a field is then optional, but more convenient than retrieving it via `getExtension(..., ...)`

Reply:
But this Endpoint is not an iq ... this is what confuses me... I was reusing what we already had there and I've added the stuff with the same logic that already existed ... Maybe you are talking about rewriting the whole ColibriConferenceIQ, which makes sense, but maybe we need to schedule this work and do it in another PR not as part of this one.

Reply:
I started looking at that on the phone and just saw the familiar getChildElementXml - and how that's built there wouldn't work if Endpoint would be an IQ, which it isn't. Sorry about the noise.